File: 1160-find-words-that-can-be-formed-by-characters-拼写单词/1160.cpp
class Solution {
public:
    int countCharacters(vector<string>& words, string chars) {
        int length = 0;
        for (const auto &word: words) {
            string temp = chars;
            bool isFormed = true;
            for (const auto &c: word) {
                int i = 0;
                for(; i != temp.size(); ++i) {
                    if (c == temp[i]) break;
                }
                if (i != temp.size()) {
                    temp.erase(i, 1);
                } else {
                    isFormed = false;
                    break;
                }
            }
            if (isFormed) {
                length += word.size();
            }
        }
        return length;
    }
};

File: 1160-find-words-that-can-be-formed-by-characters-拼写单词/1160.py
class Solution:
    def countCharacters(self, words: List[str], chars: str) -> int:
        letterTable = {}
        for c in chars:
            if c in letterTable.keys():
                letterTable[c] += 1
            else:
                letterTable[c] = 1

        rtnLen = 0
        for word in words:
            temp = letterTable.copy()
            isFormed = True
            for c in word:
                if c in temp.keys():
                    if temp[c] == 0:
                        isFormed = False
                        break
                    temp[c] -= 1
                else:
                    isFormed = False
                    break
            if isFormed:
                rtnLen += len(word)
        
        return rtnLen
